scheduler/src/cook/mesos.clj

```diff
@@ -272,6 +273,7 @@
                               ;; We will give up our leadership whenever it seems that we lost
                               ;; ZK connection
                               (when (#{ConnectionState/LOST ConnectionState/SUSPENDED} newState)
+                                (swap! mesos-leadership-atom (constantly false))
```

Comment:
From my comment above, you probably want this to just be `false`, otherwise this will evaluate to truthy